iot/api-client/beta-features/commands/receive/receive.py

```diff
@@ -0,0 +1,229 @@
+
+
+def on_connect(unused_client, unused_userdata, unused_flags, rc):
+    """Callback for when a device connects."""
+    print('on_connect', mqtt.connack_string(rc))
+
+    # After a successful connect, reset backoff time and stop backing off.
+    global should_backoff
+    global minimum_backoff_time
+    should_backoff = False
+    minimum_backoff_time = 1
+
+
+def on_disconnect(unused_client, unused_userdata, rc):
+    """Paho callback for when a device disconnects."""
+    print('on_disconnect', error_str(rc))
+
+    # Since a disconnect occurred, the next loop iteration will wait with
+    # exponential backoff.
+    global should_backoff
+    should_backoff = True
+
+
+def on_publish(unused_client, unused_userdata, unused_mid):
+    """Paho callback when a message is sent to the broker."""
+    print('on_publish')
+
+
+def on_message(unused_client, unused_userdata, message):
+    """Callback when the device receives a message on a subscription."""
+    payload = str(message.payload)
+    print('Received message \'{}\' on topic \'{}\' with Qos {}'.format(
+            payload, message.topic, str(message.qos)))
+
+
+def get_client(
+        project_id, cloud_region, registry_id, device_id, private_key_file,
+        algorithm, ca_certs, mqtt_bridge_hostname, mqtt_bridge_port):
+    """Create our MQTT client. The client_id is a unique string that identifies
+    this device. For Google Cloud IoT Core, it must be in the format below."""
+    client_id = 'projects/{}/locations/{}/registries/{}/devices/{}'.format(
+            project_id, cloud_region, registry_id, device_id)
+    client = mqtt.Client(client_id=client_id)
+
+    password = create_jwt(project_id, private_key_file, algorithm)
+
+    # With Google Cloud IoT Core, the username field is ignored, and the
+    # password field is used to transmit a JWT to authorize the device.
+    client.username_pw_set(username='unused', password=password)
+
+    # Enable SSL/TLS support.
+    client.tls_set(ca_certs=ca_certs, tls_version=ssl.PROTOCOL_TLSv1_2)
+
+    # Register message callbacks. https://eclipse.org/paho/clients/python/docs/
+    # describes additional callbacks that Paho supports. In this example, the
+    # callbacks just print to standard out.
+    client.on_connect = on_connect
+    client.on_publish = on_publish
+    client.on_disconnect = on_disconnect
+    client.on_message = on_message
+
+    # Connect to the Google MQTT bridge.
+    print('Connecting with id: {} and pass: {}'.format(client_id, password))
+    client.connect(mqtt_bridge_hostname, mqtt_bridge_port)
+
+    # This is the topic that the device will receive configuration updates on.
+    mqtt_command_topic = '/devices/{}/commands/#'.format(device_id)
+
+    # Subscribe to the config topic.
+    print('Subscribing to {}'.format(mqtt_command_topic))
+    client.subscribe(mqtt_command_topic, qos=1)
+
+    return client
+# [END iot_mqtt_config]
+
+
+def parse_command_line_args():
```

Comment:
FYI: Thea usually has me put these in the `if __name__ == "__main__"` section, but I don't have a strong opinion on this.

Reply:
Done.